clarifai/client/auth/helper.py

```diff
@@ -0,0 +1,334 @@
+          if ":" not in url:
+            raise Exception("When providing an insecure url it must have both host:port format")
+        else:
+          raise Exception("Could not get a valid response from url: %s, is the API running there?"
+                          % url) from e
+  return url
+
+
+class ClarifaiAuthHelper:
+
+  def __init__(
+      self,
+      user_id: str = "",
+      app_id: str = "",
+      pat: str = "",
+      token: str = "",
+      base: str = DEFAULT_BASE,
+      ui: str = DEFAULT_UI,
+  ):
+    """
+        A helper to get the authorization information needed to make API calls with the grpc
+        client to a specific app using a personal access token.
+
+        There are classmethods to make this object easily from either query_params provided by streamlit or from env vars.
+
+        Note: only one of personal access token (pat) or a session token (token) can be provided.
+        Always use PATs in your code and never session tokens, those are only provided internal UI code.
+
+        Args:
+          user_id: a user id who owns the resource you want to make calls to.
+          app_id: an app id for the application that owns the resource you want to interact with
+          pat: a personal access token.
+          token: a session token (internal use only, always use a PAT).
+          base: a url to the API endpoint to hit. Examples include api.clarifai.com,
+        https://api.clarifai.com (default), https://host:port, http://host:port, host:port (will be treated as http, not https). It's highly recommended to include the http:// or https:// otherwise we need to check the endpoint to determine if it has SSL during this __init__
+          ui: a url to the UI. Examples include clarifai.com,
+        https://clarifai.com (default), https://host:port, http://host:port, host:port (will be treated as http, not https). It's highly recommended to include the http:// or https:// otherwise we need to check the endpoint to determine if it has SSL during this __init__
+        """
+    if pat != "" and token != "":
+      raise Exception(
+          "A personal access token OR a session token need to be provided, but you cannot provide both."
+      )
+    elif pat == "" and token == "":
+      if os.environ.get("CLARIFAI_PAT", "") == "":
+        raise Exception("Need 'pat' in the query params or use one CLARIFAI_PAT as env var")
+      else:
+        pat = os.environ["CLARIFAI_PAT"]
+
+    self.user_id = user_id
+    self.app_id = app_id
+    self._pat = pat
+    self._token = token
+
+    self._base = https_cache(base_https_cache, base)
+    self._ui = https_cache(ui_https_cache, ui)
+
+  @classmethod
+  def from_streamlit(cls, st: Any, fallback_to_envvars: bool = True) -> "ClarifaiAuthHelper":
+    """This is a convenient method to check the query params from streamlit for the required
+        parameters for authentication, and then optional fallback to checking environment variables as
```

Comment:
For docstring convention, should we follow this below for one liner docstrings and multi-line docstrings?

One-liner:
```python3
def my_func(*args):
    """Short one liner docstrings."""
    pass
```

Multi-line docstrings:
```python3
def my_func(*args):
    """Function summary line (should be a one liner).
    
    Further function description.
    Args:
      arg_a (type):
          arg_a description.
    Returns:
      None
    """
    pass
```
The function summary line followed by a blank line below may come handy for docs incase a docstring parsers are used.